### PR TITLE
RaiPlay: Handle multiple arrays of subtitles

### PR DESCRIPTION
**Pull request by thePanz**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Bugfix
- [x] Improvement
---

### Description of your *pull request* and other information
Subtitles from RaiPlay could be defined in a new "subtitlesArray" property of the JSON.
This PR allows subtitles to be extracted from the new structure

### CODE REVIEW

File: youtube_dl/extractor/rai.py [hunk 1]
```diff
@@ -120,6 +120,19 @@ def _extract_subtitles(url, subtitle_url):
...
         return subtitles
 
+    @staticmethod
+    def _extract_subtitles_from_list(subtitle_array):
+        subtitles = {}
+        if isinstance(subtitle_array, list):
+            for item in subtitle_array:
+                lang = item.get('language')
+                url = item.get('url')
+                if isinstance(url, compat_str) and '' != url and isinstance(lang, compat_str) and '' != lang:
```
Comment (dstftw) [inline youtube_dl/extractor/rai.py]: `url_or_none`, `str_or_none`. Read coding conventions.

File: youtube_dl/extractor/rai.py [hunk 2]
```diff
@@ -142,21 +155,25 @@ class RaiPlayIE(RaiBaseIE):
...
     }, {
-        'url': 'http://www.raiplay.it/video/2014/04/Report-del-07042014-cb27157f-9dd0-4aee-b788-b1f67643a391.html',
+        'url': 'https://www.raiplay.it/video/2019/10/Report-del-21102019-La-fabbrica-della-paura-825ce3a7-8573-46c8-80d2-cde1b519fd01.html',
         'md5': '8970abf8caf8aef4696e7b1f2adfc696',
         'info_dict': {
-            'id': 'cb27157f-9dd0-4aee-b788-b1f67643a391',
-            'ext': 'mp4',
-            'title': 'Report del 07/04/2014',
-            'alt_title': 'S2013/14 - Puntata del 07/04/2014',
+            "id": "825ce3a7-8573-46c8-80d2-cde1b519fd01",
```
Comment (dstftw) [inline youtube_dl/extractor/rai.py]: Single qutoes.

File: youtube_dl/extractor/rai.py [hunk 3]
```diff
@@ -142,21 +155,25 @@ class RaiPlayIE(RaiBaseIE):
...
-            'season': '2013/14',
+            "ext": "mp4",
+            "series": "Report",
+            "season_number": 7,
+            "season": "2019/20",
+            "subtitles": {
+                "it": [
+                    {
+                        "ext": "srt",
+                        "url": "http://creativemedia4-rai-it.akamaized.net/infocdn/raitre/report/Report_EP_Puntate/11217587.srt"
```
Comment (dstftw) [inline youtube_dl/extractor/rai.py]: No exact URLs.

File: youtube_dl/extractor/rai.py [hunk 4]
```diff
@@ -191,8 +208,12 @@ def _real_extract(self, url):
         timestamp = unified_timestamp(try_get(
             media, lambda x: x['availabilities'][0]['start'], compat_str))
 
-        subtitles = self._extract_subtitles(url, video.get('subtitles'))
-
+        subtitles = {}
+        if '' != video.get('subtitles'):
+            subtitles = self._extract_subtitles(url, video.get('subtitles'))
+        else:
+            if video.get('subtitlesArray'):
```
Comment (dstftw) [inline youtube_dl/extractor/rai.py]: Both URL and array should be tried if available.

File: youtube_dl/extractor/rai.py (hunk 2 above)
Comment (thePanz) [reply youtube_dl/extractor/rai.py]: Fixed!

File: youtube_dl/extractor/rai.py (hunk 1 above)
Comment (thePanz) [reply youtube_dl/extractor/rai.py]: Fixed

File: youtube_dl/extractor/rai.py (hunk 4 above)
Comment (thePanz) [reply youtube_dl/extractor/rai.py]: Done!

File: youtube_dl/extractor/rai.py [hunk 5]
```diff
@@ -170,14 +182,13 @@ def _real_extract(self, url):
         mobj = re.match(self._VALID_URL, url)
         url, video_id = mobj.group('url', 'id')
 
-        media = self._download_json(
-            '%s?json' % url, video_id, 'Downloading video JSON')
+        media = self._download_json(url.replace('.html', '.json'), video_id, 'Downloading video JSON')
```
Comment (thePanz) [inline youtube_dl/extractor/rai.py]: The JSON url changed

**Pull request closed without merging**